### PR TITLE
Small edits to graph converter code

### PR DESCRIPTION
**Pull request by lbluque**
Simplified choosing the graph conversion function and got rid of the redundant get_neighbors method (didn't see any usage in the rest of the code)

### CODE REVIEW

File: tests/test_converter.py
Comment (BowenD-UCB) [inline tests/test_converter.py:1]: why is this removed?

Comment (lbluque) [reply tests/test_converter.py:1]: just called the method directly from `structure` instead of wrapping it in an instance method since nothing else was being done. It's just simpler and cleaner imo

Comment (BowenD-UCB) [reply tests/test_converter.py:1]: No, i was referring to the `test_crystal_graph_converter_get_neighbors():` in the test file

Comment (lbluque) [reply tests/test_converter.py:1]: Ah! because there isn't a reason to test a function from pymatgen ( it is already tested there)

Comment (BowenD-UCB) [reply tests/test_converter.py:1]: I see, that makes sense. The `converter.get_neighbor` function was originally added because I imagined there will be other `get_neighbor_list` from pymatgen in the future which not neccesarily depends only on bond distances. Removing it now is OK

File: chgnet/graph/converter.py [hunk 1]
```diff
@@ -52,17 +59,11 @@ def __init__(
...
-                self._make_graph = make_graph
-                self.create_graph = self._create_graph_fast
-                self.algorithm = "fast"
-            except (ImportError, AttributeError):
-                self.create_graph = self._create_graph_legacy
-                self.algorithm = "legacy"
-        elif algorithm == "legacy":
+        self._make_graph = None
+        if algorithm == "fast" and make_graph is not None:
+            self.create_graph = self._create_graph_fast
```
Comment (janosh) [inline chgnet/graph/converter.py:64]: We should probably print a warning or even error when `algorithm == "fast" and make_graph is None`.

Comment (lbluque) [reply chgnet/graph/converter.py]: Probably for the best. I added warnings when fast is unavailable or and unrecognised argument is passed

File: chgnet/graph/converter.py [hunk 2]
```diff
@@ -59,13 +60,24 @@ def __init__(
...
+                self.algorithm = "fast"
+            else:
+                warnings.warn(
+                    "`fast` algorithm is not available, using `legacy`",
+                    UserWarning,
+                    stacklevel=1,
+                )
+        elif algorithm != "legacy":
+            warnings.warn(
+                f"Unknown algorithm {algorithm}, using `legacy`",
```
Comment (janosh) [inline chgnet/graph/converter.py:77]: Py3.8 introduced self-documenting f-strings:

```suggestion
                f"Unknown {algorithm=}, using `legacy`",
```

Comment (lbluque) [reply chgnet/graph/converter.py]: Cool! I didn't know about this, thanks for pointing it out

**Review by BowenD-UCB — APPROVED** [review]
